### ISSUE TITLE
Reader problems after long-running

### ISSUE THREAD

**Issue by dhowe** [dhowe/ReadersJS#91]
Not sure what causes this, but things seem to go crazy if left running long enough:

![screen shot 2017-03-30 at 2 56 07 am](https://cloud.githubusercontent.com/assets/737638/24471385/8d807134-14f4-11e7-93be-74c99272e2ff.png)

**Comment by shadoof** [comment]
@dhowe this occurs on my machine after any kind of hardware? low-level? display sleep. You can reproduce it by putting your display to sleep, waiting a short while and then waking up the display. This seems 'close to the metal'. Can it be solved cross-platform? cross-browser?

**Comment by dhowe** [comment]
my guess is that draw is not being called in p5.js went the app doesn't have focus
but the time is still updating
so when you come back the fades are all expired
before finishing

so i'm guessing we need to detect that event
and either adjust the time
or simply refresh the whole page

**Comment by dhowe** [comment]
This is a default browser function; specifically that setInterval and requestAnimationFrame run slower for tabs other than the current active tab. See http://stackoverflow.com/questions/15871942/how-do-browsers-pause-change-javascript-when-tab-or-window-is-not-active

**Comment by dhowe** [comment]
I believe I've fixed this for cases where tab is not active (at least in Chrome), but not for other cases:
1. browser window does not have focus
2. display sleep
3. others?

**Comment by cqx931** [comment]
In chrome, `[UI] Window regained focus` seems to work fine when I switch around different tabs, but I rarely see `[UI] Window lost focus`, sometimes it shows up when I first start the page even though I haven't changed the tab focus yet.

**Comment by dhowe** [comment]
Right. I have commented it out in master. Leaving this part commented, can you verify which of the above cases are still problematic (on Chromium and FF) ?

**Comment by cqx931** [comment]
Case 4 is still problematic on Chrome, and opera. I'm not very sure, but the problem occurs more often when I first switch back to readersJS from another tab and quickly change the focus to another window.
Strangely, with case 4, I got no problem with Chromium so far.

In firefox, readersJS doesn't work as it does in chrome in general. I can't see so many readers simultaneously as in chrome and the speed is also very slow.

**Comment by dhowe** [comment]
We need to debug the FF problems: see #105.
Also, you are saying that Chrome/Opera have problems with case 4, but not Chromium?

**Comment by cqx931** [comment]
I was able to recreate the issue on Chromium now, then case 4 is still problematic for all chromium browsers.

**Comment by dhowe** [comment]
Right, that is why I tried onblur/onfocus, but doesn't fix the problem. But #105 is higher priority...
